src/Database/QueryBuilder.php

```diff
@@ -433,4 +434,40 @@ public function selectConcat(array $parts, string $as)
 
         return $this;
     }
+
+    /**
+     * Run a pagination count query.
+     *
+     * @param array $columns
+     * @return array
+     */
+    protected function runPaginationCountQuery($columns = ['*'])
+    {
+        if ($this->havings) {
+            $query = $this->cloneWithout(['orders', 'limit', 'offset'])
+                ->cloneWithoutBindings(['order']);
+
+            // We don't need simple columns, only specials like subselects which is why
+            // we're using havings after all.
+            foreach ($query->columns as $key => $value) {
+                if (is_string($value)) {
+                    unset($query->columns[$key]);
+                }
+            }
+
+            $countQuery = Db::connection($this->getConnection()->getName())->table(
+                Db::raw('('.$query->toSql().') as x')
```

Comment:
Any particular reason for doing `as x`? I feel like that is highly likely to conflict with developer's own custom select statements.

Reply:
No idea @LukeTowers , I just used the code @alvaro-canepa took from the other laravel package implementation.